public/app/core/services/keybindingSrv.ts

```diff
@@ -339,6 +340,14 @@ export class KeybindingSrv {
       window.location.href = window.location.href + newUrlParam;
     });
   }
+
+  static canEdit(dashboard: DashboardModel): boolean {
+    return (
+      dashboard.meta.canEdit &&
+      dashboard.meta.focusPanelId &&
+      !dashboard.panels.find(e => e.id === dashboard.meta.focusPanelId)?.repeatPanelId
```

Comment:
can you use getPanelById on daashboard model?  and not use the ? makes it a bit hard to read. Mabe this should be a function on the the DashboardModel itself, canEditPanel(panelId)   can be reused from all locations then

Reply:
apologize for be unfamiliar with code.
I have added two method `canEditPanel` and `canEditPanelById` in DashboardModel.
and use them in all branch